datalad/support/tests/test_annexrepo.py

```diff
@@ -2194,6 +2199,60 @@ def test_annexjson_protocol(path):
     assert_not_in('errors from JSON records', str(e.exception))
 
 
+@with_tempfile
+def test_annexjson_protocol_long(path):
+    # Inform about the seed here and use dedicated RNG so we are not
+    # effected by any other use of random before this test
+    seed = _seed if _seed is not None else random.randint(0, 1000)
+    # Inform for ease of debugging.
+    # TODO: if other test like this appears, craft @with_rng helper
+    print("RNG seed: %s" % seed)
+    rng = random.Random(seed)
+    records = [
+        {('field%d' % f): ("a" * rng.randint(0, 5000))
+             for f in range(rng.randint(1, 100))
+         }
+        for i in range(10)
+    ]
+    with open(path, 'w') as f:
+        for record in records:
+            print("print(%r);" % json.dumps(record), file=f)
+    runner = GitWitlessRunner()
+    res = runner.run(
+        [sys.executable, path],
+        protocol=AnnexJsonProtocol
+    )
+    eq_(res['stdout'], '')
+    eq_(res['stderr'], '')
+    eq_(res['stdout_json'], records)
+
+
+def test_annexjson_protocol_incorrect():
+    yield check_annexjson_protocol_incorrect, ''
```

Comment:
Mostly just a note for myself: This is a test of existing behavior; it passes before and after the annexrepo changes here.